performance_regression_test.py

```diff
@@ -790,3 +791,102 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+        sleep_time = 600
+        InfoEvent(message='Starting Steady State calculation for %ss' % sleep_time).publish()
+        time.sleep(sleep_time)
+        InfoEvent(message='Ended Steady State calculation.Took %ss' % sleep_time).publish()
+
+    def run_workload_and_upgrade(self, stress_cmd, sub_type=None):
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.cluster = self.db_cluster  # pylint: disable=attribute-defined-outside-init
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.tester = self  # pylint: disable=attribute-defined-outside-init
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.monitoring_set = self.monitors  # pylint: disable=attribute-defined-outside-init
+        if sub_type is None:
+            sub_type = 'read' if ' read ' in stress_cmd else 'write' if ' write ' in stress_cmd else 'mixed'
+        test_index = f'latency-during-upgrade-{sub_type}'
+        self.create_test_stats(sub_type=sub_type, append_sub_test_to_name=False, test_index=test_index)
+        stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=1, stats_aggregate_cmds=False)
+        self.steady_state_latency()
+        versions_list = []
+
+        def _get_version_and_build_id_from_node(node):
+            version = node.remoter.run('scylla --version')
+            build_id = node.remoter.run('scylla --build-id')
+            return version.stdout.strip(), build_id.stdout.strip()
+
+        for node in self.db_cluster.nodes:
+            base_version, base_build_id = _get_version_and_build_id_from_node(node)
+            self._upgrade_node(node)
+            target_version, target_build_id = _get_version_and_build_id_from_node(node)
+            versions_list.append({'base_version': base_version,
+                                  'base_build_id': base_build_id,
+                                  'target_version': target_version,
+                                  'target_build_id': target_build_id
+                                  })
+
+        # TODO: check if all `base_version` and all `target_version` are the same
+        self.update({'base_target_versions': versions_list})
+        self._stop_stress_when_finished()
+        results = self.get_stress_results(queue=stress_queue)
+        self.update_test_details(scrap_metrics_step=60)
+        self.display_results(results, test_name='test_latency_with_upgrade')
+        self.update_test_details(scrap_metrics_step=60)
+        self.display_results(results, test_name='test_latency_during_upgrade')
+        self.check_latency_during_ops(op_is_upgrade=True)
+
+    def _latency_with_upgrade(self):
+        self.run_fstrim_on_all_db_nodes()
+        self.preload_data()
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+        self.fill_db_data_for_truncate_test(insert_rows=10)
```

Comment:
Why do we need this?
Please explain in a comment.

Reply:
added

Reply:
Please open a task for such flag (which seems pretty simple to me).
and mention in the task to remove these lines when adding it.

Reply:
The "truncate during upgrade" is set as decorator during the upgrade node.
Having the truncate in this test is wrong!
The correct solution is to move the upgrade function to a shared library and use it here and in the upgrade_test where it has the decorator.